jfrog-cli/artifactory/commands/golang/go.go

```diff
@@ -114,12 +124,51 @@ func ExecuteGo(noRegistry bool, goArg, targetRepo, buildName, buildNumber string
 		if err != nil {
 			return err
 		}
+		err = goProject.LoadDependencies()
+		if err != nil {
+			return err
+		}
 		err = utils.SaveBuildInfo(buildName, buildNumber, goProject.BuildInfo(false))
 	}
 
 	return err
 }
 
+func retrieveAndPublish(targetRepo, goArg string, depsTidy bool, details *config.ArtifactoryDetails) error {
+	// Need to run Go without Artifactory to resolve all dependencies.
+	cache := goutils.GetStaticCache()
+	wd, rootDir, err := collectDependenciesPopulateAndPublish(targetRepo, depsTidy, &cache, details)
+	if err != nil {
+		if depsTidy {
+			log.Debug("Received and error:", err)
```

Comment:
I think it should be log.Error.
Also, there's a typo in the log message.
The if statement can be also simplified to avoid the *else*.

Reply:
Done